src/models/ContractEntityRepository.ts

```diff
@@ -0,0 +1,189 @@
+          allowNull: false,
+          validate: {
+            isAlphanumeric: true,
+            len: [42, 42],
+          },
+          unique: 'UK_contractaddress_entitytype',
+        },
+        entityType: {
+          type: DataTypes.ENUM({
+            values: [
+              EntityType.StakeIntentDeclared,
+              EntityType.StakeRequested,
+              EntityType.StateRootAvailable,
+              EntityType.StakeIntentConfirmed,
+              EntityType.StakeProgressed,
+              EntityType.MintProgressed,
+              EntityType.GatewayProven,
+              EntityType.BountyChangeInitiated,
+              EntityType.BountyChangeConfirmed,
+            ],
+          }),
+          unique: 'UK_contractaddress_entitytype',
+        },
+        timestamp: {
+          type: DataTypes.BIGINT,
+          allowNull: false,
+        },
+      },
+      {
+        ...initOptions,
+        modelName: 'ContractEntityModel',
+        tableName: 'contract_entity',
+      },
+    );
+  }
+
+  /**
+   * Creates a contract entity model in the repository and syncs with database.
+   * @param contractEntityAttributes ContractEntityAttributes object.
+   * @return ContractEntity object.
+   */
+  public async create(contractEntityAttributes: ContractEntityAttributes): Promise<ContractEntity> {
+    try {
+      const contractEntity: ContractEntity = await ContractEntityModel.create(
+        contractEntityAttributes,
+      ) as ContractEntity;
+      this.format(contractEntity);
+      return contractEntity;
+    } catch (e) {
+      const errorContext = {
+        attributes: contractEntityAttributes,
+        reason: e.message,
+      };
+      return Promise.reject(new Error(`Failed to create a ContractEntity: ${JSON.stringify(errorContext)}`));
+    }
+  }
+
+  /**
+   * Fetches ContractEntity data from database.
+   * @param contractEntityAttribute ContractEntityAttributes object.
+   * @returns ContractEntity object containing values which satisfy the `where` condition.
+   */
+  public async get(
+    contractEntityAttribute: ContractEntityAttributes,
+  ): Promise<ContractEntity | null> {
+    const contractEntityModel = await ContractEntityModel.findOne({
+      where: {
+        contractAddress: {
+          [Op.eq]: contractEntityAttribute.contractAddress,
+        },
+        entityType: {
+          [Op.eq]: contractEntityAttribute.entityType,
+        },
+      },
+    });
+
+    if (contractEntityModel === null) {
+      return null;
+    }
+    const contractEntity: ContractEntity = contractEntityModel;
+    this.format(contractEntity);
+    return contractEntity;
+  }
+
+  /**
+   * Updates ContractEntity data in database.
+   * @param contractEntityAttributes ContractEntityAttributes object.
+   * @return Number of affected rows.
+   */
+  public async update(contractEntityAttribute: ContractEntityAttributes): Promise<number[]> {
+    return ContractEntityModel.update(contractEntityAttribute, {
```

Comment:
As of now we need to update the timestamp. But using `contractEntityAttribute` as input will help to update more columns in future. Otherwise we would have to create different update methods for each column.